server/src/main/java/org/activityinfo/ui/client/page/config/design/ActivityForm.java

```diff
@@ -77,9 +80,17 @@ public ActivityForm(Dispatcher service, UserDatabaseDTO database) {
         add(categoryField);
 
         final MappingComboBox<Integer> locationTypeCombo = new MappingComboBox<Integer>();
-        for (LocationTypeDTO type : database.getCountry().getLocationTypes()) {
-            locationTypeCombo.add(type.getId(), type.getName());
-        }
+        database.then(new Function<UserDatabaseDTO, Object>() {
```

Comment:
@akbertram used RemoteComboBox to cover request failure in https://github.com/bedatadriven/activityinfo/commit/11ed459909fb70d0f5c5970e01563f8cfb91dc55